users/tests.py

```diff
@@ -137,10 +139,264 @@ def test_whoami(self):
+    """Tests the profile_delete endpoint. """
+
+    def test_user_profile_delete(self):
+        url = reverse('token-obtain-pair', kwargs={'version': 'v1'})
+        auth = self.client.post(url, data=json.dumps(self._user), content_type='application/json')
+        fake_header = b'Bearer ' + json.loads(auth.content)['access'].encode()
+
+        data = self._user.copy()
+        url = reverse('user-profile-delete', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_user_profile_delete_without_required_data(self):
+        url = reverse('token-obtain-pair', kwargs={'version': 'v1'})
+        auth = self.client.post(url, data=json.dumps(self._user), content_type='application/json')
+        fake_header = b'Bearer ' + json.loads(auth.content)['access'].encode()
+
+        data = {'password': self._user['password']}
+        url = reverse('user-profile-delete', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.content, b'{"username": ["This field may not be blank."]}')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        data = {'username': self._user['username']}
+        url = reverse('user-profile-delete', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.content, b'{"password": ["This field may not be blank."]}')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_user_profile_delete_with_incorrect_data(self):
+        url = reverse('token-obtain-pair', kwargs={'version': 'v1'})
+        auth = self.client.post(url, data=json.dumps(self._user), content_type='application/json')
+        fake_header = b'Bearer ' + json.loads(auth.content)['access'].encode()
+
+        data = {
+            'username': self._user['username'] * 2,
+            'password': self._user['password'] * 2,
+        }
+        response_content = (b'{"username": ["Username mismatch"], '
+                            b'"password": ["Incorrect password"]}')
+        url = reverse('user-profile-delete', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.content, response_content)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_user_profile_delete_unauthorized(self):
```

Comment:
Let's use the following docstring here: `"""Tests if it's not possible to delete the current user profile when unauthorized. """`